src/librustc_typeck/error_codes.rs

```diff
@@ -3891,6 +3891,25 @@ details.
 [issue #33685]: https://github.com/rust-lang/rust/issues/33685
 "##,
 
+E0587: r##"
+A type has conflicting packed and align representation hints.
```

Comment:
The word _conflicting_ here suggests that the hints are in conflict by value rather than inherently because they conflict by mere existence.  It may be better to simplify this sentence thusly:

```suggestion
A type has both `packed` and `align` representation hints.
```

And then allow the subsequent paragraph to explain that you can't have them